hips/hip-0016.md

```diff
@@ -0,0 +1,110 @@
+---
+hip: "0016"
+title: "Add export-values directive (similar to import-values)"
+authors: [ "Jan von Loewenstein <jan.von.loewenstein@sap.com>", "Johannes Dillmann <j.dillmann@sap.com>", "Pavel Busko <pavel.busko@sap.com>", "Philipp Stehle <philipp.stehle@sap.com>", "Ralf Pannemans <ralf.pannemans@sap.com>", "Sumit Kulhadia <sumit.kulhadia@sap.com>" ]
+created: "2022-02-01"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Composing Helm charts using dependencies is currently limited because it is not possible to create a chart with a defined set of values and distribute them to subcharts.
+
+With the `import-values` directive, there is a mechanism in Helm to use the default values of a subchart as a value in the parent chart and define the name under which the subchart value is available in the parent chart.
+This HIP proposes an `export-values` directive which is similar to the [`import-values`][import-values] directive, and allows exporting values from the parent chart to the subchart and defining the name under which it should be available.
+With that the values of the parent chart and the values of the subcharts can be defined independently. Subcharts can use their local values exclusively and don't have to be aware of the usage as a subchart. Parent charts on the other hand, can provide a consistent configuration api to their users and orchestrate the exchange of values between the subcharts using `export-values`, `import-values` or a combination thereof.
+
+This change was discussed in [helm/#3242]. There were several attempts to implement this: [helm/#7477], [helm/#10059].
+
+## Specification
+
+With the proposed `export-values` directive, users are able to specify in the **parent** chart (snippet below) which and where values should be exported to the **subchart**.
+
+```yaml
+# Chart.yaml:
+dependencies:
+  - name: client
+    version: 1.0.0
+    export-values:
+    - parent: "port"
+      child: "serverPort"
+  - name: server
+    version: 1.0.0
+    export-values:
```

Comment:
How will older versions of Helm (e.g., v3.6.2) handle running into this? What will the chart do if someone tries to install it?

Reply:
Matt good point. See my suggestion here https://github.com/helm/community/pull/242#pullrequestreview-1026427343